bukkit/src/main/java/com/github/games647/fastlogin/bukkit/task/FloodgateAuthTask.java

```diff
@@ -0,0 +1,103 @@
+package com.github.games647.fastlogin.bukkit.task;
+
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.geysermc.floodgate.api.player.FloodgatePlayer;
+
+import com.github.games647.fastlogin.bukkit.BukkitLoginSession;
+import com.github.games647.fastlogin.bukkit.FastLoginBukkit;
+import com.github.games647.fastlogin.core.StoredProfile;
+import com.github.games647.fastlogin.core.hooks.AuthPlugin;
+
+public class FloodgateAuthTask implements Runnable {
+
+    private final FastLoginBukkit plugin;
+    private final Player player;
+    private final FloodgatePlayer floodgatePlayer;
+
+    public FloodgateAuthTask(FastLoginBukkit plugin, Player player, FloodgatePlayer floodgatePlayer) {
+        this.plugin = plugin;
+        this.player = player;
+        this.floodgatePlayer = floodgatePlayer;
+    }
+
+    @Override
+    public void run() {
+        plugin.getLog().info(
+                "Player {} is connecting through Geyser Floodgate.",
+                player.getName());
+        String allowNameConflict = plugin.getCore().getConfig().getString("allowFloodgateNameConflict");
+        // check if the Bedrock player is linked to a Java account 
+        boolean isLinked = floodgatePlayer.getLinkedPlayer() != null;
+        if (allowNameConflict.equalsIgnoreCase("linked") && !isLinked) {
```

Comment:
Shouldn't we also check if the java username is the same?

Reply:
No, because you can link BE and JE accounts with different usernames.